File: The Java Program/Tokenizer.java
public class Tokenizer {

    private char[] e; //char array containing input file characters
    private int i; //index of the current character
    char currentChar; //the actual current character

    public Tokenizer(String s) {
        // constructor
        e = s.toCharArray();
        i = 0;
    }

    public Token nextToken() {
        // skip blanklike characters
        while (i < e.length && " \n\t\r".indexOf(e[i]) >= 0) {
            i++;
        }
        if (i >= e.length) {
            return new Token(TokenType.EOF);
        }
        // check for INT
        String inputString = "";
        while (i < e.length && "0123456789".indexOf(e[i]) >= 0) {
            inputString += e[i++];
        }
        if (!"".equals(inputString)) {
            return new Token(TokenType.INT, inputString);
        }
        // check for ID or reserved word        
        while (i < e.length && "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_".indexOf(e[i]) >= 0) {
            inputString += e[i++];
        }
        if (!"".equals(inputString)) {
            if ("if".equals(inputString)) {
                return new Token(TokenType.IF);
            }
            if ("else".equals(inputString)) {
                return new Token(TokenType.ELSE);
            }
            return new Token(TokenType.ID, inputString);
        }
        // We're left with strings or one character tokens
        switch (e[i++]) {
            case '{':
                return new Token(TokenType.LBRACKET,"{");              
            case '}':
                return new Token(TokenType.RBRACKET,"}");
            case '[':
                return new Token(TokenType.LSQUAREBRACKET,"[");
            case ']':
                return new Token(TokenType.RSQUAREBRACKET,"]");
            case '<':
                return new Token(TokenType.LESS,"<");
            case '>':
                return new Token(TokenType.GREATER,">");
            case '=':
                return new Token(TokenType.EQUAL,"=");
            case '"':
                String value="";
                while (i<e.length && e[i]!='"'){
                    char c=e[i++];
                    if (i>=e.length)
                        return new Token(TokenType.OTHER);
                    // check for escaped double quote
                    if (c=='\\' && e[i]=='"'){
                        c='"';
                        i++;
                    }
                    value+=c;
                } 
                i++;
                return new Token(TokenType.STRING, value);
            default:
                // OTHER should result in exception
                return new Token(TokenType.OTHER);
        }
    }
}

File: Tokenizer.php
<?php
require_once 'TokenType.php';
require_once 'Token.php';

class Tokenizer
{
    /**
     * @var: char array containing input file characters
     */
    private $e;

    /**
     * @var: Index of the current character
     */
    private $i;

    /**
     * @var: the actual current character
     */
    public $currentChar;

    /**
     * @param $s: In the input STRING file
     */
    function __construct($s){
        $this->e = str_split($s,1);
        $this->i = 0;
    }

    /**
     * To get the next token from the input string.
     * @return Token
     */
    function nextToken(){
        // Skip blanklike characters
        while ($this->i < count($this->e) && strpos(" \n\t\r", $this->e[$this->i])  !== FALSE  ){
            $this->i = $this->i + 1 ;
        }
        if ($this->i >= count($this->e) ) {
            return new Token(TokenType::$TOKENTYPE['EOF']);
        }
        // Check for INT
        $inputString = "";
        while ( $this->i < count($this->e) && strpos("0123456789",$this->e[$this->i]) !== FALSE ){
            $inputString = $inputString . $this->e[$this->i++];
        }
        if($inputString !== "")
        {
            return new Token(TokenType::$TOKENTYPE['INT'], $inputString);
        }

        // check for ID or reserved word
        while ($this->i < count($this->e) && strpos("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_", $this->e[$this->i]) !== FALSE) {
            $inputString = $inputString . $this->e[$this->i++];
        }
        if ($inputString !== ""){
            if($inputString === "if" ){
                return new Token(TokenType::$TOKENTYPE['IF']);
            }
            if($inputString === "else" ){
                return new Token(TokenType::$TOKENTYPE['ELSE']);
            }
            return new Token(TokenType::$TOKENTYPE['ID'], $inputString);
        }
        // We're left with strings or one character tokens
        switch($this->e[$this->i++])
        {
            case '{':
                return new Token(TokenType::$TOKENTYPE['LBRACKET'], "{");
            case '}':
                return new Token(TokenType::$TOKENTYPE['RBRACKET'], "}");
            case '[':
                return new Token(TokenType::$TOKENTYPE['LSQUAREBRACKET'],"[");
            case ']':
                return new Token(TokenType::$TOKENTYPE['RSQUAREBRACKET'],"]");
            case '<':
                return new Token(TokenType::$TOKENTYPE['LESS'],"<");
            case '>':
                return new Token(TokenType::$TOKENTYPE['GREATER'],">");
            case '=':
                return new Token(TokenType::$TOKENTYPE['EQUAL'],"=");
            case '"':
                $value = "";
                while($this->i < count($this->e) && $this->e[$this->i] !== '"'){
                    $c = $this->e[$this->i++];
                    if ($this->i >= count($this->e)){
                        return new Token(TokenType::$TOKENTYPE['OTHER']);
                    }
                    // check for escaped double quote
                    if($c==='\\' && $this->e[$this->i] === '"' ){
                        $c = '"';
                        $this->i++;
                    }
                    $value = $value . $c;
                }
                $this->i++;
                return new Token(TokenType::$TOKENTYPE['STRING'],$value);
            default:
                // Other Should result in exception
                return new Token(TokenType::$TOKENTYPE['OTHER']);
        }
    }
}